### DISCUSSION TITLE
Cannot use autofocus?

### DISCUSSION THREAD

**Discussion by nkgrnkgr** [vercel/next.js#17213] [Help] [answered]
# Bug report

I cannot use autofocus by using next.js

## Describe the bug

I want the page to autofocus on an input element when it loads, but it doesn't work

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. `npx create-next-app app`
2. `cd app`
3. `yarn dev`
4. write code

```html
<input type="text" autoFocus />
```
5. access `localhost:3000` and watch the input

## Expected behavior

Autofoucs input when page was loaded.

## Screenshots

<img width="1147" alt="cap1" src="https://user-images.githubusercontent.com/17094072/93649245-7e30bf00-fa46-11ea-9755-1782d4943043.png">
<img width="1136" alt="cap2" src="https://user-images.githubusercontent.com/17094072/93649251-838e0980-fa46-11ea-9746-38e67ad8b5b3.png">

## System information

- OS: macOS 10.15.6
- Browser chrome 85.0.4183.102

```json
  "dependencies": {
    "next": "9.5.3",
    "react": "16.13.1",
    "react-dom": "16.13.1"
  }
```

- node.js v14.7.0

**Comment by subha-sfcu** [comment]
Any update on this issue?

**Reply by nkgrnkgr** [reply]
Not solved yet.

**Comment by ivan-kleshnin** [comment]
Maybe posting this as an issue (not a discussion) will draw more attention?

**Comment by leerob** [comment, marked as answer]
This one tripped me up - when using `next dev`, it wasn't working as I'd expect (must be due to the browser not thinking it's a fresh page load with Fast Refresh, maybe?). When I ran `next build && next start` and also when deployed to prod, it worked!

**Reply by leerob** [reply]
You can also use React to set a ref and then call `.focus()` to programmatically focus after the first page load.

**Reply by ivan-kleshnin** [reply]
Just for the record, I often noticed that Autofocus feature doesn't work with opened devtools in Chrome.

**Reply by ypresto** [reply]
It (without `.focus()`) just works on macOS Safari 14.1.2, but not on Chrome (incl. canary 95.0.4635.2) nor Firefox 89.0.2.

**Reply by dsacramone** [reply]
I am seeing this issue in "build" mode. Any ideas? Input is losing focus when in build mode. Dev mode is fine.

**Comment by kcrwfrd** [comment]
For anyone having difficulty with this on mobile, note that iOS Safari does not support autofocus: https://caniuse.com/autofocus

iOS Safari also prevents attempts to programmatically focus the input without user interaction (e.g., on page load). See: https://gist.github.com/cathyxz/73739c1bdea7d7011abb236541dc9aaa
